src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientImpl.java

```diff
@@ -1261,35 +1267,57 @@ public void updateApplicationStaging(String applicationName, Staging staging) {
-                                                                                   .defaultIfEmpty(Collections.emptyList())
-                                                                                   .block();
-
-        List<RouteResource> outdatedRoutes = getOutdatedRoutes(appRoutes, updatedRoutes);
-        Set<CloudRoute> newRoutes = getNewRoutes(updatedRoutes, appRoutes);
-
+        List<CloudRoute> appRoutes = fetchList(() -> getRouteResourcesByAppGuid(applicationGuid), ImmutableRawCloudRoute::of);
+        Set<CloudRoute> outdatedRoutes = getOutdatedRoutes(applicationGuid, appRoutes, updatedRoutes);
+        Set<CloudRoute> newRoutes = getNewRoutes(applicationGuid, appRoutes, updatedRoutes);
         removeRoutes(outdatedRoutes, applicationGuid);
         addRoutes(newRoutes, applicationGuid);
     }
 
-    private List<RouteResource> getOutdatedRoutes(List<RouteResource> currentRoutes, Set<CloudRoute> updatedRoutes) {
-        Set<String> urls = updatedRoutes.stream()
-                                        .map(CloudRoute::getUrl)
-                                        .collect(Collectors.toSet());
+    private Set<CloudRoute> getOutdatedRoutes(UUID applicationGuid, List<CloudRoute> currentRoutes, Set<CloudRoute> updatedRoutes) {
         return currentRoutes.stream()
-                            .filter(routeResource -> !urls.contains(routeResource.getUrl()))
-                            .collect(Collectors.toList());
+                            .filter(currentRoute -> isRouteOutdated(applicationGuid, currentRoute, updatedRoutes))
+                            .collect(Collectors.toSet());
     }
 
-    private Set<CloudRoute> getNewRoutes(Set<CloudRoute> updatedRoutes, List<RouteResource> currentRoutes) {
-        Set<String> urls = currentRoutes.stream()
-                                        .map(RouteResource::getUrl)
-                                        .collect(Collectors.toSet());
+    private boolean isRouteOutdated(UUID applicationGuid, CloudRoute currentRoute, Set<CloudRoute> updatedRoutes) {
+        Optional<CloudRoute> updatedRoute = findRoute(currentRoute.getUrl(), updatedRoutes);
+        if (updatedRoute.isEmpty()) {
+            return true;
+        }
+        return isProtocolChanged(applicationGuid, currentRoute, updatedRoute.get());
+    }
+
+    private Optional<CloudRoute> findRoute(String url, Collection<CloudRoute> routes) {
+        return routes.stream()
+                     .filter(route -> Objects.equals(url, route.getUrl()))
+                     .findFirst();
+    }
+
+    private boolean isProtocolChanged(UUID applicationGuid, CloudRoute currentRoute, CloudRoute updatedRoute) {
+        if (updatedRoute.getRequestedProtocol() == null) {
+            return false;
+        }
+        return currentRoute.getDestinations()
+                           .stream()
+                           .noneMatch(routeDestination -> Objects.equals(routeDestination.getApplicationGuid(), applicationGuid)
```

Comment:
Maybe a cleaner way to do this would be 
```java
.stream()
.filter(dest -> Object.equals(dest.appGui(), appGuid))
.noneMatch(dest -> <protocol equality>)
```

Reply:
Done